src/Core/Services/Implementations/StripePaymentService.cs

```diff
@@ -377,21 +366,6 @@ private async Task ChangeOrganizationSponsorship(Organization org, OrganizationS
 
         if (user.Gateway == GatewayType.Stripe && !string.IsNullOrWhiteSpace(user.GatewayCustomerId))
         {
-            if (!string.IsNullOrWhiteSpace(paymentToken))
-            {
-                try
-                {
-                    await UpdatePaymentMethodAsync(user, paymentMethodType, paymentToken, true, taxInfo);
```

Comment:
🤔  It's not immediately clear to me why this is getting deleted. What's your thought behind it?

Reply:
The way I read this, since this code block explicitly calls `UpdatePaymentMethodAsync` with `allowInAppPurchases = true`, it's only relevant for those types. I can see how this might be unsafe however and other payment updates with a token -- are there any? -- might still need this, but then should the exception be suppressed?

Reply:
I went through and was able to reproduce this. It seems that that flag is only used to throw the `GatewayException`. This code is still hit when a user who previously paid for premium outside of an organization lets their subscription cancel, then resubscribes after it's already cancelled. In that case, the `user.GatewayCustomerId` will still have a value, and the `paymentToken` will be the payment method ID for the card that they set up on the second signup. All that to say, this should probably stay, just sans the boolean!

Reply:
OK, lemme know if https://github.com/bitwarden/server/pull/3640/commits/83509538b2777585b9755c2b76c828b2b0c44621 works then.